suseconnect/migration.go

```diff
@@ -0,0 +1,653 @@
+	//go:embed migrationUsage.txt
+	migrationUsageText string
+	// flag indicating interuption by INT/TERM signal
+	interrupted bool
+
+	// ErrInterrupted is returned when execution was interrupted by INT/TERM signal
+	ErrInterrupted = errors.New("Interrupted")
+)
+
+// logger shortcuts
+var (
+	Debug      *log.Logger = connect.Debug
+	QuietOut   *log.Logger = connect.QuietOut
+	VerboseOut             = log.New(io.Discard, "", 0)
+)
+
+// this Value type is used to hold values of args which could be passed multiple times
+// e.g. $ command -x 1 -x 2 -x 3
+type multiArg []string
+
+func (a *multiArg) String() string {
+	return strings.Join(*a, "|")
+}
+func (a *multiArg) Set(v string) error {
+	*a = append(*a, v)
+	return nil
+}
+
+func migrationMain() {
+	var (
+		// dummy flag to keep default but accept cli arg
+		dummy          bool
+		debug          bool
+		verbose        bool
+		quiet          bool
+		nonInteractive bool
+		noSnapshots    bool
+		noSelfUpdate   bool
+		migrationNum   int
+		fsRoot         string
+		from           multiArg
+		repo           multiArg
+		download       multiArg // using multiArg here to make flags simpler to visit
+	)
+
+	flag.Usage = func() {
+		fmt.Print(migrationUsageText)
+	}
+	flag.BoolVar(&debug, "debug", false, "")
+	flag.BoolVar(&dummy, "no-verbose", false, "")
+	flag.BoolVar(&verbose, "verbose", false, "")
+	flag.BoolVar(&verbose, "v", false, "")
+	flag.BoolVar(&dummy, "no-quiet", false, "")
+	flag.BoolVar(&quiet, "quiet", false, "")
+	flag.BoolVar(&quiet, "q", false, "")
+	flag.BoolVar(&nonInteractive, "non-interactive", false, "")
+	flag.BoolVar(&nonInteractive, "n", false, "")
+	flag.BoolVar(&noSnapshots, "no-snapshots", false, "")
+	flag.BoolVar(&dummy, "selfupdate", false, "")
+	flag.BoolVar(&noSelfUpdate, "no-selfupdate", false, "")
+	flag.IntVar(&migrationNum, "migration", 0, "")
+	flag.StringVar(&fsRoot, "root", "", "")
+	// zypper dup passthrough args
+	// bool flags don't need variables as these will be processed using flag.Visit()
+	flag.BoolVar(&dummy, "auto-agree-with-licenses", false, "")
+	flag.BoolVar(&dummy, "l", false, "")
+	flag.BoolVar(&dummy, "allow-vendor-change", false, "")
+	flag.BoolVar(&dummy, "no-allow-vendor-change", false, "")
+	flag.BoolVar(&dummy, "debug-solver", false, "")
+	flag.BoolVar(&dummy, "recommends", false, "")
+	flag.BoolVar(&dummy, "no-recommends", false, "")
+	flag.BoolVar(&dummy, "replacefiles", false, "")
+	flag.BoolVar(&dummy, "details", false, "")
+	flag.BoolVar(&dummy, "download-only", false, "")
+	flag.Var(&download, "download", "")
+	flag.Var(&from, "from", "")
+	flag.Var(&repo, "r", "")
+	flag.Var(&repo, "repo", "")
+
+	flag.Parse()
+	if err := checkFlagContradictions(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// this is only to keep the flag parsing logic simple and avoid double
+	// negations/negatives below
+	selfUpdate := !noSelfUpdate
+
+	if verbose {
+		VerboseOut.SetOutput(os.Stdout)
+	}
+
+	if debug {
+		connect.EnableDebug()
```

Comment:
Maybe instead of providing this option, we could just always configure `connect.Debug` to append to `/var/log/zypper-migration.log`. Then if something goes wrong, support should be able to determine why from the details of the zypper and api calls.